x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/execution/TriggeredWatchStore.java

```diff
@@ -112,12 +156,15 @@ private BulkRequest createBulkRequest(final List<TriggeredWatch> triggeredWatche
         return request;
     }
 
+    /**
+     * Delete a triggered watch entry.
+     * Note that this happens asynchronously, as these kind of requests are batched together to reduce the amount of concurrent requests.
+     *
+     * @param wid The ID os the triggered watch id
+     */
     public void delete(Wid wid) {
         DeleteRequest request = new DeleteRequest(TriggeredWatchStoreField.INDEX_NAME, TriggeredWatchStoreField.DOC_TYPE, wid.value());
-        try (ThreadContext.StoredContext ignore = stashWithOrigin(client.threadPool().getThreadContext(), WATCHER_ORIGIN)) {
-            client.delete(request); // FIXME shouldn't we wait before saying the delete was successful
-        }
-        logger.trace("successfully deleted triggered watch with id [{}]", wid);
+        bulkProcessor.add(request);
```

Comment:
Does executing watches take into consideration any deleted requests here? what happens if i delete a watch and then wait to submit the bulk, might that watch trigger again? Should we make deletes be the only sync thing? I think thats a fair tradeoff given my very little knowledge here.

Reply:
yes it does. If the watch is deleted and then a triggered watch is picked, there will be a watch history entry telling you that the referenced watch could not be found.